sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1412,11 +1470,13 @@ def stop_worker(self):
 
 
 class WorkerHandlerManager(object):
-  def __init__(self, environments, job_provision_info):
+  def __init__(self, environments, job_provision_info, state_cache_size):
     self._environments = environments
     self._job_provision_info = job_provision_info
     self._cached_handlers = collections.defaultdict(list)
-    self._state = FnApiRunner.StateServicer() # rename?
+    self._state = sdk_worker.CachingMaterializingStateHandler(
+        StateCache(state_cache_size),
```

Comment:
For my learning, why is the cache required inside the fn_api_runner vs. on the SDK side?

Reply:
Good catch. This does not seem to be necessary. Let me revisit.

Reply:
I added this because the WorkerHandlerManager will insert the state handler into the WorkerHandlerFactory, which generates a cached BundleProcessorCache with that state handler for the EmbeddedWorkerHandler. It is not necessary otherwise and just does unnecessary caching on the FnApiRunner side.

Reply:
Fixed now.